quil-py/src/program/calibration.rs

```diff
@@ -0,0 +1,134 @@
+use pyo3::prelude::*;
+use quil_rs::{
+    expression::Expression,
+    instruction::{
+        Calibration, GateModifier, Instruction, MeasureCalibrationDefinition, Measurement, Qubit,
+    },
+    program::CalibrationSet,
+};
+use rigetti_pyo3::{
+    impl_as_mut_for_wrapper, impl_repr, py_wrap_type, PyTryFrom, PyWrapper, PyWrapperMut, ToPython,
+    ToPythonError,
+};
+
+use crate::instruction::{
+    PyCalibration, PyExpression, PyGateModifier, PyInstruction, PyMeasureCalibrationDefinition,
+    PyMeasurement, PyQubit,
+};
+
+use super::ProgramError;
+
+py_wrap_type! {
+    PyCalibrationSet(CalibrationSet) as "CalibrationSet"
+}
+impl_as_mut_for_wrapper!(PyCalibrationSet);
+impl_repr!(PyCalibrationSet);
+
+#[pymethods]
+impl PyCalibrationSet {
+    #[new]
+    pub fn new(
+        py: Python<'_>,
+        calibrations: Vec<PyCalibration>,
+        measure_calibrations: Vec<PyMeasureCalibrationDefinition>,
+    ) -> PyResult<Self> {
+        Ok(Self(CalibrationSet {
+            calibrations: Vec::<_>::py_try_from(py, &calibrations)?,
+            measure_calibrations: Vec::<_>::py_try_from(py, &measure_calibrations)?,
+        }))
+    }
+
+    #[getter]
+    pub fn calibrations(&self, py: Python<'_>) -> PyResult<Vec<PyCalibration>> {
+        self.as_inner().calibrations().to_python(py)
+    }
+
+    #[getter]
+    pub fn measure_calibrations(
+        &self,
+        py: Python<'_>,
+    ) -> PyResult<Vec<PyMeasureCalibrationDefinition>> {
+        self.as_inner().measure_calibrations().to_python(py)
+    }
+
+    pub fn expand(
```

Comment:
I'll make an issue for this:

The ability to expand an entire program, given that `quil-rs` can already do this and it meshes with the content of this MR. This functionality will be needed by some consumers of pyQuil, who currently do that iteration & expansion within their own programs; the v4 changes _may_ make that harder but a method on `Program` could do it for them.

Reply:
Makes sense! Thanks for the issue.